modules/rarity/rarity.gd

```diff
@@ -0,0 +1,53 @@
+extends Node
+
+
+#region Public Variables
+var rarities: Array[StringName] = []
+var rarity_colors: Dictionary = {}
+#endregion
+
+
+#region Internal Functions
+func _ready() -> void:
+	Modules.register_hooks(handler)
+	
+	register_rarity(&"Free", Color.WHITE)
+	register_rarity(&"Common", Color.GRAY)
+	register_rarity(&"Rare", Color.BLUE)
+	register_rarity(&"Epic", Color.PURPLE)
+	register_rarity(&"Legendary", Color.GOLD)
```

Comment:
We register all rarities in the base rarity module since individual rarities aren't unique enough to warrent their own seperate modules.